### PR TITLE
fix: fully compatiable with original menus/component design

### LINKED ISSUE #218: fully compatiable with original menus/component design
## Scenarios

To be fully compatible with previous kubebb component design,we need :

- keep `parentOwnerReferences` in `menus` untile auto menus supported in `kubebb`  (this requirement comes from https://github.com/kubebb/core/discussions/200#discussioncomment-6633538)
- add a new CRD `Portal` in kubebb to configure `portal path`

For this new CRD `Portal`:
```yaml
type PortalSpec struct {
   Entry string
   Path string
}
```

## Sample
Take bc-apis as an example https://github.com/kubebb/components/blob/main/charts/bc-apis/templates/menus.yaml
1. Menus
only update apiVersion from `component.t7d.io/v1beta1` to `core.kubebb.k8s.com.cn/v1alpha1`

2. Component
replace [`component.yaml`](https://github.com/kubebb/components/blob/main/charts/bc-apis/templates/component.yaml) to `portal.yaml`

The `portal.yaml` would be :
```yaml
apiVersion: core.kubebb.k8s.com.cn/v1alpha1
kind: Portal
metadata:
  name: bc
spec:
  entry: /bc-public/index.html
  path: /bc
```

### PR DESCRIPTION
**Pull request by bjwswang**
…st during component migration

#### What type of PR is this?
/kind revert
/kind feature

#### What this PR does / why we need it
See https://github.com/kubebb/core/issues/218
#### Which issue(s) this PR fixes

Fixes #218

### CODE REVIEW

File: config/samples/kustomization.yaml [hunk 1]
```diff
@@ -4,4 +4,5 @@ resources:
 - core_v1alpha1_subscription.yaml
 - core_v1alpha1_componentplan.yaml
 - core_v1alpha1_component.yaml
+- core_v1alpha1_portal.yaml
```
Comment (Abirdcfly) [inline config/samples/kustomization.yaml:7]: should git add this file?

**Review by Abirdcfly — APPROVED** [review]

**Review by 0xff-dev — APPROVED** [review]

**Comment by bjwswang** [comment]
@bobsongplus @Carrotzpc any comments on these changes ?

File: config/rbac/portal_viewer_role.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# permissions for end users to view portals.
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: portal-viewer-role
```
Comment (Carrotzpc) [inline config/rbac/portal_viewer_role.yaml:5]: 所有用户都需要 portal 资源的查看权限，是不是得把这个 role 跟之前 component 资源的那个 role 做一样的处理？

Comment (bjwswang) [reply config/rbac/portal_viewer_role.yaml:5]: are u saying this field? 
https://github.com/kubebb/components/blob/main/charts/u4a-component/templates/bff-server/rbac.yaml#L50

Comment (bjwswang) [reply config/rbac/portal_viewer_role.yaml:5]: will update the rbac  in components later

File: api/v1alpha1/portal_webhook.go [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+//+kubebuilder:webhook:path=/validate-core-kubebb-k8s-com-cn-v1alpha1-portal,mutating=false,failurePolicy=fail,sideEffects=None,groups=core.kubebb.k8s.com.cn,resources=portals,verbs=create;update,versions=v1alpha1,name=vportal.kb.io,admissionReviewVersions=v1
+
+var _ webhook.CustomValidator = &Portal{}
+
+// ValidateCreate implements webhook.Validator so a webhook will be registered for the type
+func (r *Portal) ValidateCreate(ctx context.Context, obj runtime.Object) error {
+	portallog.Info("validate create", "name", r.Name)
+
+	// TODO: validate entry & path conflicts
```
Comment (bjwswang) [inline api/v1alpha1/portal_webhook.go:45]: FIXME later. Will need to validate entry & path conflicts later. May need control.client

**Comment by bjwswang** [comment]
i will merge this pr and leave `webhook implemetation` to next PR

**Pull request merged**